worker/backend/backend.go

```diff
@@ -106,14 +113,75 @@ func (b *Backend) Create(gdnSpec garden.ContainerSpec) (container garden.Contain
 //
 // Errors:
 // * TODO.
-func (b *Backend) Destroy(handle string) (err error) { return }
+func (b *Backend) Destroy(handle string) error {
+	ctx := namespaces.WithNamespace(context.Background(), b.namespace)
+	const maxTaskKillWaitTime = 10 * time.Second
+
+	if handle == "" {
+		return InputValidationError{}
+	}
+
+	container, err := b.client.GetContainer(ctx, handle)
+	if err != nil {
+		return err
+	}
+
+	task, err := container.Task(ctx, nil)
+	if err != nil {
+		if !errdefs.IsNotFound(err) {
+			return err
+		}
+
+		return b.client.Destroy(ctx, handle)
+	}
+
+	timeDelimitedContext, _ := context.WithTimeout(ctx, maxTaskKillWaitTime)
+	err = killTasks(timeDelimitedContext, task)
+	if err != nil {
+		return err
+	}
+
+	return b.client.Destroy(ctx, handle)
+}
```

Comment:
what about to refactor the logic as below (move `killTasks` into else branch, to have less return route path):
```
	task, err := container.Task(ctxWithTimeout, nil)
	
	if err != nil { //error occurs
		if !errdefs.IsNotFound(err) { //a real error but the task can not be found
			return ClientError { InnerError: err }
		}
                // do nothing if we could not find the task
	} else {
               // kill the task
		err = killTasks(ctxWithTimeout, task)
		if err != nil {
			return ClientError{ InnerError: err }
		}
	}
        // no matter the task exists or is killed, the container should be destroyed anyway. 
	err = b.client.Destroy(ctxWithTimeout, handle)
	if err != nil {
		return ClientError{ InnerError: err }
	}
```